pkg/ccl/storageccl/export_storage_test.go

```diff
@@ -120,38 +127,83 @@ func TestPutHttp(t *testing.T) {
-		switch r.Method {
-		case "PUT":
-			f, err := os.Create(localfile)
-			if err != nil {
-				http.Error(w, err.Error(), 500)
-				return
-			}
-			defer f.Close()
+	makeServer := func() (*url.URL, func() int, func()) {
+		var files int
+		srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 			defer r.Body.Close()
-			if _, err := io.Copy(f, r.Body); err != nil {
-				http.Error(w, err.Error(), 500)
-				return
+			localfile := filepath.Join(tmp, filepath.Base(r.URL.Path))
+			switch r.Method {
+			case "PUT":
+				f, err := os.Create(localfile)
+				if err != nil {
+					http.Error(w, err.Error(), 500)
+					return
+				}
+				defer f.Close()
+				defer r.Body.Close()
+				if _, err := io.Copy(f, r.Body); err != nil {
+					http.Error(w, err.Error(), 500)
+					return
+				}
+				files++
+			case "GET":
+				http.ServeFile(w, r, localfile)
+			case "DELETE":
+				if err := os.Remove(localfile); err != nil {
+					http.Error(w, err.Error(), 500)
+					return
+				}
+			default:
+				http.Error(w, "unsupported method "+r.Method, 400)
 			}
-		case "GET":
-			http.ServeFile(w, r, localfile)
-		case "DELETE":
-			if err := os.Remove(localfile); err != nil {
-				http.Error(w, err.Error(), 500)
-				return
-			}
-		default:
-			http.Error(w, "unsupported method "+r.Method, 400)
+		}))
+		t.Logf("Mock HTTP Storage %q", srv.URL)
+		uri, err := url.Parse(srv.URL)
+		if err != nil {
+			srv.Close()
+			t.Fatal(err)
+		}
+		return uri, func() int { return files }, srv.Close
+	}
+
+	t.Run("singleHost", func(t *testing.T) {
+		srv, files, cleanup := makeServer()
+		defer cleanup()
+		testExportToTarget(t, roachpb.ExportStorage{
+			Provider: roachpb.ExportStorageProvider_Http,
+			HttpPath: roachpb.ExportStorage_Http{BaseUri: srv.String() + "/"},
+		})
+		if wrote := files(); wrote != 11 {
+			t.Fatal("expected 11 files to be written to single http store")
+		}
+	})
+
+	t.Run("multiHost", func(t *testing.T) {
+		s1, f1, cleanup1 := makeServer()
+		defer cleanup1()
+		s2, f2, cleanup2 := makeServer()
+		defer cleanup2()
+		s3, f3, cleanup3 := makeServer()
+		defer cleanup3()
+
+		combined := *s1
+		combined.Host = strings.Join([]string{s1.Host, s2.Host, s3.Host}, ",")
+
+		testExportToTarget(t, roachpb.ExportStorage{
+			Provider: roachpb.ExportStorageProvider_Http,
+			HttpPath: roachpb.ExportStorage_Http{BaseUri: combined.String() + "/"},
+		})
+		// testExportToTarget writes 10 simple files and one big one, so expect 11.
+		w1, w2, w3 := f1(), f2(), f3()
+		if expected, actual := 11, w1+w2+w3; expected != actual {
+			t.Fatalf("expected %d files written across all hosts, got %d", expected, actual)
+		}
+		if expected, actual := 3, w1; expected != actual {
+			t.Fatalf("expected %d files written to http host 1, got %d", expected, actual)
+		}
+		if expected, actual := 4, w3; expected != actual {
+			t.Fatalf("expected %d files written to http host 3, got %d", expected, actual)
```

Comment:
And this could avoid the `w`s entirely with

```go
if expected, actual := 3, files1(); expected != actual {
  t.Fatalf("expected %d files written to http host 1, got %d", expected, actual)
}
if expected, actual := 3, files2(); expected != actual {
  t.Fatalf("expected %d files written to http host 2, got %d", expected, actual)
}
if expected, actual := 4, files3(); expected != actual {
  t.Fatalf("expected %d files written to http host 3, got %d", expected, actual)
}
```